xblockutils/studio_editable.py

```diff
@@ -0,0 +1,295 @@
+        #     data["name"] = data["name"].strip()
+        pass
+
+    def validate_field_data(self, validation, data):
+        """
+        Validate this block's field data. Instead of checking fields like self.name, check the
+        fields set on data, e.g. data.name. This allows the same validation method to be re-used
+        for the studio editor. Any errors found should be added to "validation".
+
+        This method should not return any value or raise any exceptions.
+        All of this XBlock's fields should be found in "data", even if they aren't being changed
+        or aren't even set (i.e. are defaults).
+        """
+        # Example:
+        # if data.count <=0:
+        #     validation.add(ValidationMessage(ValidationMessage.ERROR, u"Invalid count"))
+        pass
+
+    def validate(self):
+        """
+        Validates the state of this XBlock.
+
+        Subclasses should override validate_field_data() to validate fields and override this
+        only for validation not related to this block's field values.
+        """
+        validation = super(StudioEditableXBlockMixin, self).validate()
+        self.validate_field_data(validation, self)
+        return validation
+
+
+class StudioContainerXBlockMixin(object):
+    """
+    An XBlock mixin to provide convenient use of an XBlock in Studio
+    that wants to allow the user to assign children to it.
+    """
+    has_author_view = True  # Without this flag, studio will use student_view on newly-added blocks :/
+
+    def render_children(self, context, fragment, can_reorder=True, can_add=False):
+        """
+        Renders the children of the module with HTML appropriate for Studio. If can_reorder is
+        True, then the children will be rendered to support drag and drop.
+        """
+        contents = []
+
+        for child_id in self.children:
+            child = self.runtime.get_block(child_id)
+            if can_reorder:
+                context['reorderable_items'].add(child.scope_ids.usage_id)
+            rendered_child = child.render('author_view' if hasattr(child, 'author_view') else 'student_view', context)
+            fragment.add_frag_resources(rendered_child)
+
+            contents.append({
+                'id': child.location.to_deprecated_string(),
+                'content': rendered_child.content
+            })
+
+        fragment.add_content(self.runtime.render_template("studio_render_children_view.html", {
+            'items': contents,
+            'xblock_context': context,
+            'can_add': can_add,
+            'can_reorder': can_reorder,
+        }))
+
+    def author_view(self, context):
+        """
+        Display a the studio editor when the user has clicked "View" to see the container view,
+        otherwise just show the normal 'author_preview_view' or 'student_view' preview.
+        """
+        root_xblock = context.get('root_xblock')
+        is_root = root_xblock and root_xblock.location == self.location
```

Comment:
Nit: I'm unclear why you need to define this variable if you're only using it in the next line. I'd err on the side of just doing, `if root_xblock and root_xblock.location == self.location:` - it's easier to read.